lib/src/utf8.dart

```diff
@@ -31,15 +31,16 @@ class Utf8 extends Struct {
 
   /// Creates a [String] containing the characters UTF-8 encoded in [string].
   ///
-  /// The [string] must be a zero-terminated byte sequence of valid UTF-8
-  /// encodings of Unicode scalar values. A [FormatException] is thrown if the
-  /// input is malformed. See [Utf8Decoder] for details on decoding.
+  /// Either the [string] must be zero-terminated or its [length] -- the number
+  /// of bytes -- must be specified as a non-negative value. The byte sequence
```

Comment:
You can replace ` -- ` by `&mdash;` to get proper em-dashes around the parenthetical phrase.

Reply:
Any preference on "null byte" & "null-terminated" vs. "zero byte" & "zero-terminated"? Looks like I accidentally made it inconsistent because the others are all "null".

Reply:
The "null terminated" refers to the character (ASCII NUL) and "zero terminated" refers to the byte.
Since value below 128 are both bytes and characters in UTF-8, it's a mostly philosophical question whether you consider the string terminated by the zero byte before decoding it or the NUL character after decoding it.

We do find the length *before* decoding, and it's possible to have an overlong encoding of the NUL character (even if it's invalid), and we won't terminate the string on that, so "zero byte terminated" seems most precise.

(I'd not use `null byte`. Either null/NULL/NUL *character* or zero *byte*.)